pegasus_wrapper/workflow.py

```diff
@@ -247,6 +249,20 @@ def write_dax_to_dir(self, output_xml_dir: Optional[Path] = None) -> Path:
         if not output_xml_dir:
             output_xml_dir = self._workflow_directory
 
+        num_jobs = len(self._job_graph.jobs.keys())
+        num_ckpts = len([ckpt_file for ckpt_file in output_xml_dir.rglob("___ckpt")])
+        if num_jobs == num_ckpts:
```

Comment:
This check won't work quite right if we replace some of the jobs (say the very last jobs -- the ones that nothing else depends on) and then try to rerun. That shouldn't result in a no-op if the names are different. On the other hand this may be good enough? I wouldn't expect that to come up too often.

Generally I think this is going to be fragile when you are developing a workflow. If job names are changing somewhat regularly but you keep the same workflow directory, this may get triggered spuriously.

If it's too hard to check this "the right way" then maybe just reword/add to the message. I think changing "will" to "might/may" is enough.

Reply:
If I'm not mistaken, job names are largely just dependent on the experiment_name + locator of the job in question. Therefore, job names I can't imagine changing all that much even if they are now doing completely different things. I do see what you're saying though and this is fragile. I will probably leave a comment in there and change the "will" to "might/may". 

Lastly though: as a frequent user, do you imagine this being more annoying than helpful in your development process? I'm including this as a convenience, so if it actually is the opposite, I can keep it on my local only.

Reply:
It depends a lot on how you create your locators, I guess.

I think this will be more helpful than annoying. It seems like a win to save yourself from accidentally creating a no-op workflow, running it, and then wondering why nothing seems to have been executed while you were gone (or worse, not realizing nothing was updated). I don't think it will come up often enough to be too annoying.